### PR TITLE
Add `is_z_forward` option to `looking_at()` & `look_at()` and some constants for glTF space

### PR DESCRIPTION
**Pull request by TokageItLab**
By convention in OpenGL, objects are often imported facing the camera. This convention has been brought to glTF, so Godot imports glTF assets with +Z as forward.

In environments where this convention exists, there should be two `look_at()`s, one for the camera and one for the asset.

However, since Godot considers all coordinate systems to be unified by the camera space, so the `look_at()` and the `Vector3` orientation constants are correct for the camera, but not for the glTF asset.

This PR adds some options that assume an imported glTF with +Z as forward. Those should be methods that users have already implemented as the workarounds, it is just not supported by Godot in the core.

In fact, this option is only needed for those that use `look_at()` and have the forward axis hard-coded, so most methods and classes do not need it and core compatibility is fully preserved.

The only change is regarding PathFollow3D, as the default direction should be the opposite.

### CODE REVIEW

**Comment by QbieShay** [comment]
I don't think this is the right solution, like mentioned yesterday we need to discuss and come to a consensus before implementing this.

**Comment by TokageItLab** [comment]
Yes, so it is still a 4.x milestone, just same as #72753 PR.

**Comment by lyuma** [comment]
To be clear, is_gltf would be set for FBX and blend files as well, since those are converted through gltf.

Does this affect OBJ and DAE? What cases is the is_gltf option intended to cover?

Also, the description mentions cameras, but cameras can be imported from gltf as well.

**Comment by TokageItLab** [comment]
Indeed, since the imported glTF camera will be a Godot camera, `is_gltf` may be not good. So `is_z_forward` seems to be common and better.

File: core/variant/variant_call.cpp [hunk 1]
```diff
@@ -2512,6 +2512,12 @@ static void _register_variant_builtin_methods() {
 	_VariantCall::add_variant_constant(Variant::VECTOR3, "DOWN", Vector3(0, -1, 0));
 	_VariantCall::add_variant_constant(Variant::VECTOR3, "FORWARD", Vector3(0, 0, -1));
 	_VariantCall::add_variant_constant(Variant::VECTOR3, "BACK", Vector3(0, 0, 1));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_LEFT", Vector3(1, 0, 0));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_RIGHT", Vector3(-1, 0, 0));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_UP", Vector3(0, 1, 0));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_DOWN", Vector3(0, -1, 0));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_FORWARD", Vector3(0, 0, 1));
+	_VariantCall::add_variant_constant(Variant::VECTOR3, "GLTF_BACK", Vector3(0, 0, -1));
```
Comment (aaronfranke) [inline core/variant/variant_call.cpp:2520]: I am in favor of the `looking_at` changes, but not these. Perhaps this should be a separate PR? I think if we had a PR just for the looking_at changes this would be easier to approve.

File: scene/2d/path_2d.cpp [hunk 2]
```diff
@@ -315,8 +315,8 @@ void PathFollow2D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_rotates", "enable"), &PathFollow2D::set_rotates);
 	ClassDB::bind_method(D_METHOD("is_rotating"), &PathFollow2D::is_rotating);
 
-	ClassDB::bind_method(D_METHOD("set_cubic_interpolation", "enable"), &PathFollow2D::set_cubic_interpolation);
-	ClassDB::bind_method(D_METHOD("get_cubic_interpolation"), &PathFollow2D::get_cubic_interpolation);
+	ClassDB::bind_method(D_METHOD("set_cubic_interpolation", "enabled"), &PathFollow2D::set_cubic_interpolation);
+	ClassDB::bind_method(D_METHOD("is_cubic_interpolation"), &PathFollow2D::is_cubic_interpolation);
```
Comment (aaronfranke) [inline scene/2d/path_2d.cpp:319]: Similar thoughts here, this seems unrelated, I would split into multiple PRs.

These renames also break compatibility, so it can't be done for 4.1. If you want to see these renames in 4.0 I suggest you make a PR with only these renames which can be merged in advance of the other changes. Otherwise this will have to wait for 5.0.

Comment (TokageItLab) [reply scene/2d/path_2d.cpp]: Separated as #72842 and #72843.

File: core/variant/variant_call.cpp (hunk 1 above)
Comment (TokageItLab) [reply core/variant/variant_call.cpp:2520]: I suppose this is necessary as a justification for the `is_z_forward` option. The documentation also describes glTF forward (+Z). 

The +Z concept remains when it comes to importing/exporting assets. For example, it feels strange to specify `BACK` if you want the model to face forward for retargeting/exporting glTF.

**Comment by TokageItLab** [comment]
Separated the fix for PathFollow. It should be merged by 4.0.

And this PR will need to be merged after reaching consensus after 4.0.

File: core/variant/variant_call.cpp (hunk 1 above)
Comment (aaronfranke) [reply core/variant/variant_call.cpp:2520]: I disagree, it's not tied to glTF. Here's a PR with just the look_at changes: https://github.com/godotengine/godot/pull/75875

Comment (TokageItLab) [reply core/variant/variant_call.cpp:2520]: > Or you could have -Z forward in glTF if you are controlling a glTF camera which has its forward pointing -Z.

Well, even if the enum GLTF_FORWARD is an exaggeration, it is a definition that all of the assets as a GLTF are +Z, and a GLTF with -Z as the front is not a GLTF. A GLTF with a -Z front is not a GLTF. It is what is called a fake GLTF. GLTF cameras are out of scope here.

There may be a better enum (e.g. ASSET_FORWARD), but even if you can't find it, you must document somewhere that as long as Godot uses the GLTF importer for all formats and has a GLTF exporter, the asset must be +Z oriented. You have to document somewhere that the asset needs to be +Z oriented. (Root motion is also needed that document.)

**Comment by TokageItLab** [comment]
Superseded by #76060
